### PR TITLE
add link check  to markdown recipe

### PR DESCRIPTION
**Pull request by katyamust**
[#477]
Add markdown link checking to the markdown recipe

### CODE REVIEW

File: code-reviews/recipes/Markdown.md [hunk 1]
```diff
@@ -85,17 +85,19 @@ npx write-good *.md
 
 Write Good is also available as an [extension for VS Code](https://marketplace.visualstudio.com/items?itemName=travisthetechie.write-good-linter)
 
-## VS Code Extensions
+### VS Code Extensions
```
Comment (TessFerrandez) [inline code-reviews/recipes/Markdown.md:88]: This should stay a level 2 title to follow the same format as the other recipes

Comment (TessFerrandez) [inline code-reviews/recipes/Markdown.md:88]:
```suggestion
## VS Code Extensions
```

File: code-reviews/recipes/Markdown.md [hunk 2]
```diff
@@ -85,17 +85,19 @@ npx write-good *.md
...
 
-### Write Good Linter
+#### Write Good Linter
```
Comment (TessFerrandez) [inline code-reviews/recipes/Markdown.md:90]:
```suggestion
### Write Good Linter
```

File: code-reviews/recipes/Markdown.md [hunk 3]
```diff
@@ -85,17 +85,19 @@ npx write-good *.md
...
 
 The [`Write Good Linter Extension`](https://marketplace.visualstudio.com/items?itemName=travisthetechie.write-good-linter) integrates with VS Code to give grammar and language advice while editing the document.
 
-### markdownlint extension
+#### markdownlint extension
```
Comment (TessFerrandez) [inline code-reviews/recipes/Markdown.md:94]:
```suggestion
### markdownlint extension
```

File: .github/workflows/markdownlint.yml [hunk 4]
```diff
@@ -28,4 +26,3 @@ jobs:
       with:
         use-quiet-mode: 'yes'
         use-verbose-mode: 'no'
```
Comment (TessFerrandez) [inline .github/workflows/markdownlint.yml:28]: you need to keep the config-file: this contains the exceptions like ./link-to-the-work-item - this is why you are getting the link error in your PR.

**Review by TessFerrandez — CHANGES_REQUESTED** [review]

File: code-reviews/recipes/Markdown.md (hunk 1 above)
Comment (katyamust) [reply code-reviews/recipes/Markdown.md:88]: I have got impression by reading the document that we are talking about linting tools here and VS code extensions are one of those tools - hence the change to level of title. Did I mistunderstood? @TessFerrandez

File: .github/workflows/markdownlint.yml [hunk 5]
```diff
@@ -4,12 +4,10 @@ on:
   pull_request:
     branches:
       - master
-
+
```
Comment (tompaana) [inline .github/workflows/markdownlint.yml:7]:
```suggestion

```

File: code-reviews/recipes/Markdown.md [hunk 6]
```diff
@@ -85,17 +85,19 @@ npx write-good *.md
...
 
 The [`markdownlint extension`](https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint) examines the markdown documents, showing warnings for rule violations while editing.
 
-## Build validation
+## Build Validation
+
+### linting
```
Comment (tompaana) [inline code-reviews/recipes/Markdown.md:100]:
```suggestion
### Linting
```

File: code-reviews/recipes/Markdown.md [hunk 7]
```diff
@@ -127,6 +129,20 @@ jobs:
         markdownlint "**/*.md" --ignore node_modules
 ```
 
+### link check
```
Comment (tompaana) [inline code-reviews/recipes/Markdown.md:132]:
```suggestion
### Checking Links
```

**Review by tompaana — CHANGES_REQUESTED** [review]

File: .github/workflows/markdownlint.yml (hunk 4 above)
Comment (katyamust) [reply .github/workflows/markdownlint.yml:28]: Yepp, reverted back

**Review by tompaana — APPROVED** [review]
I see my suggested changes have been implemented. Approving on my behalf. Thank you!

File: code-reviews/recipes/Markdown.md (hunk 1 above)
Comment (TessFerrandez) [reply code-reviews/recipes/Markdown.md:88]: I think we should be consistent between recipes, the VS code extensions could be linting tools, they could also be other extensions that are useful - e.g. viewers, or in this case the write good extension etc. that are not necessarily linters

Comment (katyamust) [reply code-reviews/recipes/Markdown.md]: Yes, that sound good then,  updated.

**Review by TessFerrandez — APPROVED** [review]

**Comment by TessFerrandez** [comment]
@katyamust can you add the link that is failing to the exclude list? It should technically not fail the link check but seems to fail for all PRs for some reason so better to add it to the exclude list

**Comment by katyamust** [comment]
> @katyamust can you add the link that is failing to the exclude list? It should technically not fail the link check but seems to fail for all PRs for some reason so better to add it to the exclude list

I have done that @TessFerrandez , thanks for tip. That link was actually working for me so I was not sure what to do. Now there are number of other dead links. This is becoming an issue I am fixing those almost every time  I submit something. 

May be we should tackle this differently and run link check only on timely basis. Lets discuss.

**Pull request merged**